app/assets/javascripts/discourse/app/lib/wait-for-keyboard.js

```diff
@@ -0,0 +1,66 @@
+import { getOwner } from "@ember/application";
+
+export async function waitForKeyboard(context) {
```

Comment:
This is just closing, right? Not opening?
```suggestion
export async function waitForKeyboardToClose(context) {
```

Reply:
No it's both, it's mostly "waitForKeyboardChangeState"

Reply:
Ah I see... but does that mean we're waiting the full 1s to open a modal if the keyboard is already closed? 🤔

Reply:
It shouldnt because:

```
    if (!document.documentElement.classList.contains("keyboard-visible")) {
      return resolve({ visible: false });
    }
```

Reply:
We have

```
if (!document.documentElement.classList.contains("keyboard-visible")) {
  return resolve({ visible: false });
}
```
which would return immediately if the keyboard is not visible. So I think that means this function only works to wait for closing?

Reply:
Yes I see what you mean, it's slightly confusing indeed, and the function name is not totally valid

Reply:
I think I have no choice to have two functions then, a `waitForOpen` and a `waitForClose`